logstash-core/lib/logstash/codecs/delegator.rb

```diff
@@ -14,6 +16,8 @@ def metric=(metric)
       __getobj__.metric = metric
 
       __getobj__.metric.gauge(:name, __getobj__.class.config_name)
+      __getobj__.metric.gauge("parent-code-ref", @parent_code_reference)
```

Comment:
Nitpick: other gauge id's are symbols, so providing these as symbols will reduce the likelihood of confusion
```suggestion
      __getobj__.metric.gauge(:"parent-code-ref", @parent_code_reference)
```